src/scripts/get-temporary-user-password.ts

```diff
@@ -0,0 +1,82 @@
+import { Argv } from 'yargs';
+import { RootCommand } from '..';
+
+import { OktaUserService, User, UserService } from './services/user-service';
+import { oktaManageClient } from './services/client-service';
+import * as TE from 'fp-ts/lib/TaskEither';
+import * as E from 'fp-ts/lib/Either';
+import * as O from 'fp-ts/lib/Option';
+import { flow, pipe } from 'fp-ts/lib/function';
+import * as Console from 'fp-ts/lib/Console';
+
+export const getTemporaryUserPassword = (
+  service: UserService,
+  userId: string
+): TE.TaskEither<
+  string,
+  {
+    readonly user: User;
+    readonly temporaryPassword: string;
+  }
+> =>
+  pipe(
+    userId,
+    service.getUser,
+    TE.chain(
+      flow(
+        O.fold(
+          () =>
+            TE.left(
+              `User [${userId}] does not exist. Cannot get temporary user password.`
+            ),
+          // eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
+          () => service.getTemporaryUserPassword(userId)
+        )
+      )
+    ),
+    TE.chainFirstIOK(({ user, temporaryPassword }) =>
+      Console.info(
+        `Retrieved temporary password for user [${user.id}] [${user.email}]. Temporary password is [${temporaryPassword}].`
+      )
+    )
+  );
+
+export default (
+  // eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
+  rootCommand: RootCommand
+): Argv<{
+  readonly clientId: string;
+  readonly privateKey: string;
+  readonly organisationUrl: string;
+  readonly userId: string;
+}> =>
+  rootCommand.command(
+    'get-temporary-user-password [user-id]',
```

Comment:
**suggestion:** even though this matches the Okta API, I think it is worth breaking naming alignment here. The important bit here isn't `getting` the temporary password. It is expiring the existing password and setting a new one.

Can you think of a better name for this command?

Reply:
actually that was a name i came up with. okta called it expirePasswordAndGetTemporaryPassword in their api. Should i rename it to this instead?

Reply:
expirePassword can be done with or without getting temporary password. I think "expirePasswordAndGetTemporaryPassword" fits well. I just tried to shorten it earlier

Reply:
modified in commit: e3747a4725da06fa2ac00eb63f1015edda638086